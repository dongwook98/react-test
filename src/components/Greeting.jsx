/* eslint-disable react/prop-types */
const Greeting = ({ name }) => {
  return <h1>Hello, {name}</h1>;
};

export default Greeting;
