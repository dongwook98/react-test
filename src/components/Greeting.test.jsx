/* eslint-disable no-undef */
import { render } from '@testing-library/react';
import { describe, expect } from 'vitest';
import Greeting from './Greeting';

describe('Greeting Component', () => {
  it('should render correctly', () => {
    const { getByText } = render(<Greeting name={'Alice'} />);
    expect(getByText('Hello, Alice')).toBeInTheDocument();
  });

  // 스냅샷 테스트: 컴포넌트가 의도대로 렌더링되는지 확인
  // 단축키 u로 스냅샷 업데이트 해줘야함
  it('should render', () => {
    const { asFragment } = render(<Greeting name={'Bob'} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
