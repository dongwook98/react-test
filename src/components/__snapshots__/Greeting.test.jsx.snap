// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Greeting Component > should render 1`] = `
<DocumentFragment>
  <h1>
    Hello, Bob
  </h1>
</DocumentFragment>
`;
